openmmforcefields/generators/template_generators.py

```diff
@@ -1289,14 +1289,19 @@ def __init__(self, molecules=None, cache=None, forcefield=None):
 
         # Create ForceField object
         import openff.toolkit.typing.engines.smirnoff
-        try:
-            filename = forcefield
-            if not filename.endswith('.offxml'):
-                filename += '.offxml'
+
+        # check for an installed force field
+        available_force_fields = openff.toolkit.typing.engines.smirnoff.get_available_force_fields()
+        if (filename := forcefield + ".offxml") in available_force_fields or (filename := forcefield) in available_force_fields:
             self._smirnoff_forcefield = openff.toolkit.typing.engines.smirnoff.ForceField(filename)
-        except Exception as e:
-            _logger.error(e)
-            raise ValueError(f"Can't find specified SMIRNOFF force field ({forcefield}) in install paths") from e
+
+        # just try parsing the input and let openff handle the error
+        else:
+            try:
+                self._smirnoff_forcefield = openff.toolkit.typing.engines.smirnoff.ForceField(forcefield)
+            except Exception as e:
+                _logger.error(e)
+                raise ValueError(f"Can't find specified SMIRNOFF force field ({forcefield}) in install paths") from e
```

Comment:
I think we want to improve the error message here, since as far as I can see it's not now that we don't find the ff on the install paths but it's rather that the provided string is not correctly formatted as a ffxml string.

Is it too crazy to add some more logic here such that we can have specific error messages as to what is causing this? It doesn't seem too hard to accomplish, as far as I can tell.

Reply:
I have updated the error to reflect that a missing file or poorly formatted string could cause this error. As for specific messages on the formatting of the forcefield I would prefer to leave this to the `openff.toolkit.ForceField` and just raise the error from there as it's quite descriptive itself.